### PR TITLE
[docs] landing page (fixes #21750)

### LINKED ISSUE #21750: [docs] Improve the Getting Started section
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

Expand on what we have now, but go much deeper / be more helpful. Answer what Ray is, why you need it, where to learn more and how to get started. This is where new users land.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by maxpumperla**
Preview: [docs](https://ray--21859.org.readthedocs.build/en/21859/)

More informative and attractive landing page with a focus on

- [x] Three core personas with clearer messaging
- [x] A better overview of what Ray is and what you can do with it
- [x] More instructive "get involved"
- [x] Adding a "what docs are for you?" guide

Screenshot 2022-01-25 at 16 40 49

### CODE REVIEW

**Comment by maxpumperla** [comment]
@richardliaw @ericl @edoakes wording matters here, please give me all your nitpicks.

**Comment by richardliaw** [comment]
A couple stylistic changes to request:

Screen Shot 2022-01-25 at 11 17 54 AM

Can we avoid overlapping into the right sidebar?
---------------

Screen Shot 2022-01-25 at 11 18 40 AM
Can we make the buttons consistent?

File: doc/source/index.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+As a general-purpose and universal distributed compute framework, you can flexibly run any compute-intensive Python workload — from distributed training or hyperparameter tuning to deep reinforcement learning and production model serving.
+To help you, Ray Core provides a simple, universal API for building distributed applications.
+And Ray's native libraries and tools enable you to run complex ML applications with Ray.
+You can deploy these applications on any of the major cloud providers, including AWS, GCP, and Azure, or run them on your own servers.
+Ray also has a growing [ecosystem of community integrations](ray-overview/ray-libraries), including [Dask](https://docs.ray.io/en/latest/data/dask-on-ray.html), [MARS](https://docs.ray.io/en/latest/data/mars-on-ray.html), [Modin](https://github.com/modin-project/modin), [Horovod](https://horovod.readthedocs.io/en/stable/ray_include.html), [Hugging Face](https://huggingface.co/transformers/main_classes/trainer.html#transformers.Trainer.hyperparameter_search), [Scikit-learn](ray-more-libs/joblib), [and others](ray-more-libs/index).
+The following figure gives you an overview of the Ray ecosystem.
```
Comment (richardliaw) [inline doc/source/index.md:90]: Right now this reads as a big paragraph; can we insert some linespaces?

File: doc/source/index.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+:card:
+
+**Build distributed applications with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">Core**
+^^^
+Ray Core provides a [simple and flexible API](ray-core/walkthrough.rst) for building and running your distributed applications.
+You can often parallelize single machine code with little to zero code changes.
+
+For instance, with Ray Core you can easily parallelize your functions using 
+[Ray Tasks](ray-core/walkthrough.rst#remote-functions-tasks) or your classes using 
+[Ray Actors](ray-core/actors.rst).
```
Comment (richardliaw) [inline doc/source/index.md:37]:
```suggestion
Ray Core provides a [simple and flexible API](ray-core/walkthrough.rst) for building and running your distributed applications.
You can easily parallelize your functions using 
[Ray Tasks](ray-core/walkthrough.rst#remote-functions-tasks) or your classes using 
[Ray Actors](ray-core/actors.rst). 
```

File: doc/source/index.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
++++
+```{link-button} ray-core/walkthrough
+:type: ref
+:text: Get Started with Ray Core
+:classes: btn-outline-info btn-block
+```
+---
+**Run machine learning workflows with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">ML**
+^^^
+With Ray ML you can run tasks like [distributed data loading and compute](data/dataset.rst),
```
Comment (richardliaw) [inline doc/source/index.md:47]:
```
Ray ML is a toolkit for distributed machine learning. It provides libraries for distributed data processing, model training, tuning, reinforcement learning, model serving, and more. 
```

Perhaps we can add links to each, but I think we should condense the wording

File: doc/source/index.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+```{link-button} ray-overview/index
+:type: ref
+:text: Get Started with Ray ML
+:classes: btn-outline-info btn-block
+```
+
+---
+**Deploy large-scale workloads with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">Cluster**
+^^^
+With Ray Cluster you can deploy your workloads on [AWS, GCP, Azure](cluster/quickstart) or
```
Comment (richardliaw) [inline doc/source/index.md:65]:
```suggestion
With a Ray cluster, you can deploy your workloads on [AWS, GCP, Azure](cluster/quickstart) or 
```

File: doc/source/index.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
++++
+```{link-button} ray-overview/index
+:type: ref
+:text: Get Started with Ray ML
+:classes: btn-outline-info btn-block
+```
+
+---
+**Deploy large-scale workloads with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">Cluster**
```
Comment (richardliaw) [inline doc/source/index.md:63]:
```suggestion
**Deploy large-scale workloads with <img src="ray-overview/images/ray_svg_logo.svg" alt="ray" width="50px">Clusters**
```

Comment (richardliaw) [reply doc/source/index.md:63]: Also consider removing the svg? Not sure if it scales well across devices.

File: doc/source/index.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+You can also use [Ray Cluster Managers](cluster/deploy) to run Ray on your existing
+[Kubernetes](cluster/kubernetes),
+[YARN](cluster/yarn),
+or [Slurm](cluster/slurm) clusters.
++++
+
+```{link-button} cluster/quickstart
+:type: ref
+:text: Get Started with Ray Cluster
```
[the remark this answers is not in the document]
Comment (richardliaw) [reply doc/source/index.md:77]: I'm not sure about this. Can you give feedback on this particular element @edoakes @ericl ?

File: doc/source/index.md (hunk 5 above)
Comment (maxpumperla) [reply doc/source/index.md:61]: It's the size of the font, which does not change. can remove it later if people hate it.

**Comment by maxpumperla** [comment]
@richardliaw I addressed your comments and added a getting started guide today. This should be good to go now.

Screenshot 2022-01-26 at 15 53 36
 
Screenshot 2022-01-26 at 15 53 46

File: doc/source/_toc.yml [hunk 7]
```diff
@@ -99,32 +100,19 @@ parts:
       - file: ray-core/examples/overview
         title: "Tutorials and Examples"
 
-  - caption: Deploying Ray Clusters
+  - caption: Ray Clusters | Deployments
```
Comment (richardliaw) [inline doc/source/_toc.yml:103]: FYI, the caption says "Clusters" here but elsewhere we have "Ray Cluster", should we make that consistent?

**Review by richardliaw — APPROVED** [review]
Looks good to me!

**Review by ericl — APPROVED** [review]

**Pull request merged**